src/renderer/containers/ReplayFileStats/PunishTable.tsx

```diff
@@ -0,0 +1,292 @@
+import { ConversionType, StatsType, StockType } from "@slippi/slippi-js";
+import { FileResult } from "../../../common/replayBrowser/types";
+import React from "react";
+import { convertFrameCountToDurationString } from "../../../common/time";
+import * as T from "./TableStyles";
+import _ from "lodash";
+import { getCharacterIcon } from "@/lib/utils";
+import { extractPlayerNames } from "common/matchNames";
+
+const columnCount = 6;
+
+export interface PunishTableProps {
+  file: FileResult;
+  stats: StatsType | null;
+  playerIndex: number;
+}
+
+export const PunishTable: React.FC<PunishTableProps> = ({
+  file,
+  stats,
+  playerIndex,
+}) => {
+  if (!stats) return <div>An Error Occurred!</div>;
+  const toOrdinal = (i: number) => {
+    const j = i % 10,
+      k = i % 100;
+    if (j == 1 && k != 11) {
+      return i + "st";
+    }
+    if (j == 2 && k != 12) {
+      return i + "nd";
+    }
+    if (j == 3 && k != 13) {
+      return i + "rd";
+    }
+    return i + "th";
+  };
```

Comment:
This should probably be an exported function in like `lib` somewhere rather than inside the component.